syz-fuzzer/testing.go

```diff
@@ -148,7 +149,39 @@ func checkMachine(args *checkArgs) (*rpctype.CheckArgs, error) {
 	if err := checkSimpleProgram(args, features); err != nil {
 		return nil, err
 	}
-	return checkCalls(args, features)
+	res, err := checkCalls(args, features)
+	if err != nil {
+		return nil, err
+	}
+	log.Logf(0, "checking modules...")
+	var addr uint64
+	res.Modules = append(res.Modules, rpctype.KernelModule{
+		Name: "",
+		Addr: 0,
+	})
+	kallsyms, _ := ioutil.ReadFile("/proc/kallsyms")
+	if len(kallsyms) > 0 {
+		re := regexp.MustCompile(`([a-fA-F0-9]+) T _text\n`)
+		if m := re.FindSubmatch(kallsyms); m != nil {
+			if _, err := fmt.Sscanf(string(m[1]), "%x", &addr); err != nil {
+				return nil, fmt.Errorf("_text parsing error in /proc/kallsyms")
+			}
+		}
+		res.Modules[0].Addr = addr
```

Comment:
I assumed that kernel address is 0 and suggested to remove some if's that avoided adding kernel address to symbols.
But it turns out to be non-0. So do we need to add it or not (removing the if's changed logic)?
We already properly handled everything related to the main kernel, no? Do we need to parse /proc/kallsyms at all? What did not we handle correctly?

Reply:
res.Modules[0].Addr = addr
here the addr is _text, it's needed also do these if's